lightning/src/chain/chainmonitor.rs

```diff
@@ -306,12 +315,17 @@ impl<ChannelSigner: Sign, C: Deref, T: Deref, F: Deref, L: Deref, P: Deref> even
 	      L::Target: Logger,
 	      P::Target: channelmonitor::Persist<ChannelSigner>,
 {
-	fn get_and_clear_pending_events(&self) -> Vec<Event> {
+	/// Processes [`SpendableOutputs`] events produced from each [`ChannelMonitor`] upon maturity.
+	///
+	/// [`SpendableOutputs`]: events::Event::SpendableOutputs
```

Comment:
Should we note explicitly that calling back into the provider is OK?

Reply:
Hmm, good catch. Though we don't expect any need to call back, It's actually not safe to call `watch_channel` since it would attempt to take a `write` lock while already holding a `read` lock.

Reply:
Actually, never mind. We drop the read lock before calling the handler.